packages/cdktf-cli/src/lib/local-provider-versions.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) HashiCorp, Inc
+// SPDX-License-Identifier: MPL-2.0
+import fs from "fs-extra";
+import path from "path";
+import { CdktfConfig } from "./cdktf-config";
+import { Errors } from "./errors";
+import { logger } from "./logging";
+
+/**
+ * Class to help with reading `versions.json` file
+ * published by `provider get` or `provider add` commands
+ */
+export class LocalProviderVersions {
+  private versions: Record<string, string> | undefined;
+
+  private get jsonFilePath(): string {
+    const config = CdktfConfig.read();
+    return path.resolve(config.codeMakerOutput, "versions.json");
+  }
+
+  private async readLocalVersionsJson(): Promise<void> {
+    let versionsJson;
+    try {
+      versionsJson = await fs.readFile(this.jsonFilePath, "utf8");
+    } catch (e) {
+      // If we cannot read the versions.json file, it may not exist
+      logger.debug("versions.json file reading error: ", e);
+      this.versions = undefined;
+      return;
+    }
+
+    let providerVersions;
+    try {
+      providerVersions = JSON.parse(versionsJson) as LocalProviderVersions;
+    } catch (e) {
+      throw Errors.External(
+        "versions.json file is malformed. The root must be a JSON object."
+      );
+    }
+
+    this.versions = Object.fromEntries(
+      Object.entries(providerVersions).map(([providerFqn, versions]) => {
+        return [
+          providerFqn
+            .replace("registry.terraform.io/", "")
+            .replace("hashicorp/", "")
+            .toLowerCase(),
```

Comment:
nit: While I don't think that this changes – we do have constants for those:
https://github.com/hashicorp/terraform-cdk/blob/4f03ceb074c9710a406da5cd90d86c927ce6e5ca/packages/cdktf-cli/src/lib/dependencies/dependency-manager.ts#L20